source/common/http/header_map_impl.cc

```diff
@@ -19,10 +19,15 @@ constexpr size_t MinDynamicCapacity{32};
 // This includes the NULL (StringUtil::itoa technically only needs 21).
 constexpr size_t MaxIntegerLength{32};
 
+size_t newCapacity(uint32_t existing_capacity, uint32_t size_to_append) {
+  return (static_cast<size_t>(existing_capacity) + size_to_append) * 2;
+}
+
 void validateCapacity(size_t new_capacity) {
   // If the resizing will cause buffer overflow due to hitting uint32_t::max, an OOM is likely
   // imminent. Fast-fail rather than allow a buffer overflow attack (issue #1421)
-  RELEASE_ASSERT(new_capacity <= std::numeric_limits<uint32_t>::max(), "");
+  RELEASE_ASSERT(new_capacity <= std::numeric_limits<uint32_t>::max(),
```

Comment:
std::numeric_limits<T>::max() returns type T; should that part be cast to size_t before comparison?